force-app/main/default/lwc/newServiceSchedule/newServiceSchedule.js

```diff
@@ -153,7 +206,39 @@ export default class NewServiceSchedule extends LightningElement {
             (endTime.getMinutes() - startTime.getMinutes()) / 60;
     }
 
+    setFirstSessionStartTimeAndEndTime(startDate) {
+        if (!startDate) {
+            return;
+        }
+
+        let startTime = new Date(startDate);
+        let endTime = new Date(startDate);
+
+        this.dateFields.start.value = new Date(startTime).toISOString();
+        endTime.setHours(startTime.getHours() + this.duration);
+        endTime.setMinutes(startTime.getMinutes() + (this.duration % 1) * 60);
+        this.dateFields.end.value = new Date(endTime).toISOString();
+    }
+
     get disableSessionEnd() {
         return !this.dateFields.start.value;
     }
+
+    validateServiceScheduleOnOrAfter() {
+        let noError = true;
+
+        if (this.picklistFields.frequency.value === ONE_TIME) {
+            return noError;
+        }
+
+        noError =
+            (this.template.querySelector('[data-element="seriesEndsOn"]') === null ||
```

Comment:
This looks great but these large conditionals can sometimes be hard to follow. What do you think about breaking this down, maybe checking the value of  the seriesEnd picklist field and if it is ON then make sure seriesEndsOn has a value else make sure that number of sessions has a value?

Reply:
I had done it that way but I thought you would not like it so did it this way. I can change it